app/javascript/Pages/Login.jsx

```diff
@@ -1,64 +1,79 @@
-import React from "react";
+import React, { Component } from "react";
 import { Inertia } from "@inertiajs/inertia";
-import Layout from "./Layout";
+import Navbar from "./components/Navbar/Navbar";
 import { Head, Link } from "@inertiajs/inertia-react";
+import "../stylesheets/body.css";
+import "../stylesheets/login.css";
 
-export default function Login(props) {
-  const ErrorMessage = () => {
-    if (props.error) {
-      return <p>{props.error}</p>;
-    }
-    return null;
-  };
+class Login extends Component {
+  render() {
+    const ErrorMessage = () => {
+      if (this.props.error) {
+        return <p>{this.props.error}</p>;
+      }
+      return null;
+    };
 
-  const handleSubmit = (e) => {
-    const token = document.querySelector(
-      "[name=csrf-token]"
-    ).content;
+    const handleSubmit = (e) => {
+      const token = document.querySelector("[name=csrf-token]").content;
 
-    const data = {
-      username: username.value,
-      password: password.value,
-    };
+      const data = {
+        username: username.value,
+        password: password.value,
+      };
 
-    e.preventDefault();
-    Inertia.post("/login", data, {
-      headers: {
-        "X-CSRF-Token": token,
-      },
-    });
-  };
+      e.preventDefault();
```

Comment:
**suggestion (blocking)**: preventDefault should be the first thing you do in the event handler because there might be cases where the event already continued if you do it later.